Modules/_datetimemodule.c

```diff
@@ -40,6 +40,27 @@
 
 #define PyTimezone_Check(op) PyObject_TypeCheck(op, &PyDateTime_TimeZoneType)
 
+typedef struct {
+    /* Conversion factors. */
+    PyObject *us_per_ms;       // 1_000
+    PyObject *us_per_second;   // 1_000_000
+    PyObject *us_per_minute;   // 1e6 * 60 as Python int
+    PyObject *us_per_hour;     // 1e6 * 3600 as Python int
+    PyObject *us_per_day;      // 1e6 * 3600 * 24 as Python int
+    PyObject *us_per_week;     // 1e6 * 3600 * 24 * 7 as Python int
+    PyObject *seconds_per_day; // 3600 * 24 as Python int
+
+    /* The interned UTC timezone instance */
+    PyObject *utc;
+
+    /* The interned Epoch datetime instance */
+    PyObject *epoch;
+} datetime_state;
+
+static datetime_state _datetime_global_state;
+
+#define GLOBAL_STATE() (&_datetime_global_state)
```

Comment:
I suppose that you will soon move it somewhere else, not make it "global", but maybe "per interpreter". I suggest a name which doesn't say "global".

I suggest: "get_datetime_state()".

It may make following changes shorter.

Reply:
Well, I disagree; I think it is important to communicate that this is not module state. Perhaps the macro can be named `STATIC_STATE` instead. It communicates status quo perfectly:

- state is stored in a static global variable
- state is fetched using a macro

Reply:
Ok, as you want. It was just a suggestion.